packages/kbn-dev-cli-runner/src/run.ts

```diff
@@ -31,18 +31,24 @@ export interface RunOptions {
   description?: string;
   log?: {
     defaultLevel?: LogLevel;
+    context?: string;
   };
   flags?: FlagOptions;
 }
 
 export async function run<T>(fn: RunFn<T>, options: RunOptions = {}): Promise<T | undefined> {
   const flags = getFlags(process.argv.slice(2), options.flags, options.log?.defaultLevel);
-  const log = new ToolingLog({
-    level: pickLevelFromFlags(flags, {
-      default: options.log?.defaultLevel,
-    }),
-    writeTo: process.stdout,
-  });
+  const log = new ToolingLog(
```

Comment:
On the side, I've extended the tooling logger with an optional context. If you think this is too much noise in this PR, I can move this out to a different

Reply:
I think it would be good to split the context out since it's a decent chunk of the PR and gives separation of concerns.

Reply:
works both ways for me 😃 It's not in the main context of the PR but also not too big to void the possibility of being included here. Maybe a good reason to separate from this PR is to later reference, or pinpoint in case we need

Reply:
I've extracted this to https://github.com/elastic/kibana/pull/190766

Reply:
👍